keel-ec2-plugin/src/main/kotlin/com/netflix/spinnaker/keel/api/ec2/Scaling.kt

```diff
@@ -17,10 +17,170 @@
  */
 package com.netflix.spinnaker.keel.api.ec2
 
+import com.fasterxml.jackson.annotation.JsonIgnore
 import com.fasterxml.jackson.annotation.JsonInclude
 import com.fasterxml.jackson.annotation.JsonInclude.Include.NON_EMPTY
+import de.danielbechler.diff.inclusion.Inclusion
+import de.danielbechler.diff.introspection.ObjectDiffProperty
+import java.time.Duration
 
+val DEFAULT_AUTOSCALE_INSTANCE_WARMUP: Duration = Duration.ofMinutes(5)
+
+@JsonInclude(NON_EMPTY)
 data class Scaling(
-  @JsonInclude(NON_EMPTY)
-  val suspendedProcesses: Set<ScalingProcess> = emptySet()
+  val suspendedProcesses: Set<ScalingProcess> = emptySet(),
+  val targetTrackingPolicies: Set<TargetTrackingPolicy> = emptySet(),
+  val stepScalingPolicies: Set<StepScalingPolicy> = emptySet()
```

Comment:
What’s the purpose of separating these things? Can a cluster have both?

Reply:
A cluster can have both; that's actually the common case at Netflix - our core team recommends using target tracking policies to scale out (with scaleIn disabled) and a step policy to scale-in. They have different models and incompatible methods of application. It's like amazon's classic "elb" vs. application and network load balancers, where plenty of backwards incompatible api improvements were made in the release of the newer implementation.